### PR TITLE
ci(github-workflows)[OK-88] Add a step for veryfing deps;  refactor

### PR DESCRIPTION
**Pull request by Wajktor13**
This PR:
- refactors github workflow files
- adds a step for verifying deps to frontend and backend workflows
- fixes imports
- removes unused deps

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -46,7 +46,6 @@
     "@nestjs/mongoose": "^10.0.6",
     "@nestjs/passport": "^10.0.3",
     "@nestjs/platform-express": "^10.0.0",
-    "@nestjs/schematics": "^10.0.0",
```
Comment (marcepanowyy) [inline backend/package.json]: schematics is useful for creating components in CLI (services, modules etc.)
```
nest g mo users
```

File: backend/package.json [hunk 2]
```diff
@@ -60,6 +59,7 @@
     "cache-manager": "^5.7.3",
     "class-transformer": "^0.5.1",
     "class-validator": "^0.14.1",
+    "express": "^4.19.2",
```
Comment (marcepanowyy) [inline backend/package.json]: express is by default built in nestjs so no need to install

File: backend/package.json [hunk 3]
```diff
@@ -74,16 +74,11 @@
...
     "eslint": "^8.57.0",
-    "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-nestjs": "^1.2.3",
-    "eslint-plugin-prettier": "^5.0.0",
     "jest": "^29.5.0",
     "prettier": "^3.0.0",
-    "source-map-support": "^0.5.21",
     "supertest": "^7.0.0",
     "ts-jest": "^29.1.0",
-    "ts-loader": "^9.4.3",
```
Comment (marcepanowyy) [inline backend/package.json]: ts-loader might be needed to handle TS files and convert it into JS files

**Comment by Wajktor13** [comment]
@Strattoss @Adam0s007 check if I haven't removed something that is needed, please.

**Review by marcepanowyy — APPROVED** [review]

**Review by adbista — APPROVED** [review]
It seems to be working for me

File: backend/.depcheckrc [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ignores: ["@nestjs/schematics", "ts-loader"]
```
Comment (marcepanowyy) [inline backend/.depcheckrc]: I would revert some packages - they are useful. They were added by default when project was created.
```
ignores: ["@nestjs/schematics", "ts-loader", "source-map-support", "tsconfig-paths"]
```

File: backend/package.json [hunk 5]
```diff
@@ -74,16 +74,12 @@
...
-    "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-nestjs": "^1.2.3",
-    "eslint-plugin-prettier": "^5.0.0",
     "jest": "^29.5.0",
     "prettier": "^3.0.0",
-    "source-map-support": "^0.5.21",
     "supertest": "^7.0.0",
     "ts-jest": "^29.1.0",
     "ts-loader": "^9.4.3",
-    "ts-node": "^10.9.1",
```
Comment (marcepanowyy) [inline backend/package.json]: .

File: backend/package.json [hunk 6]
```diff
@@ -74,16 +74,12 @@
...
-    "tsconfig-paths": "^4.2.0"
```
Comment (marcepanowyy) [inline backend/package.json]: .

**Review by marcepanowyy — APPROVED** [review]

**Pull request merged**